tests/common/runners/test_runnable.py

```diff
@@ -47,11 +48,11 @@ def test_runnable_direct_worker_call() -> None:
 
 def test_fail_on_process_worker_started_early() -> None:
     # process pool cannot be started before class instance is created: mapping not exist in worker
-    with Pool(4) as p:
+    with ProcessPoolExecutor(4) as p:
         r = _TestRunnableWorkerMethod(4)
         with pytest.raises(KeyError):
             r._run(p)
-        p.close()
+        p.shutdown(wait=True)
```

Comment:
@rudolfix This test fails, doesn't raise and I'm not totally clear on what it's doing. Looks like the executor initializes the process pool lazily on first tasks, so maybe this order doesn't matter now?

Reply:
if processes are lazily instantiated then yes. just remove the `with raises` and test if the run was successful.